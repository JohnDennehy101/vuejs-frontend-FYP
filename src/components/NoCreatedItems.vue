<template>
  <div id="no-items-call-to-action-container">
    <i class="fas fa-folder-open"></i>
    <h4>{{ messageForUser }}</h4>

    <button @click="$router.push({ path: `${this.link}` })">
      {{ callToActionForUser }}
    </button>
  </div>
</template>

<script>
export default {
  props: ["display", "message", "callToAction", "routerLink"],
  data() {
    return {
      messageForUser: this.message,
      callToActionForUser: this.callToAction,
      link: this.routerLink,
      noItemsPresent: this.display,
    };
  },
};
</script>

<style scoped lang="scss">
#no-items-call-to-action-container {
  height: 20vh;
  padding: 1rem 0;
  width: 60%;
  border-radius: 10px;
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 1px solid #eeeeee;
  background-color: white;
}
button {
  border-radius: 10px;
  border: none;
  width: 40%;
  height: 2.5rem;
  margin: 1rem 0;
  background-color: $primary-button-background-colour;
  color: $primary-button-text-colour;
}
button:hover {
  cursor: pointer;
}
svg {
  font-size: 1.5rem;
  padding: 1rem;
}
</style>
