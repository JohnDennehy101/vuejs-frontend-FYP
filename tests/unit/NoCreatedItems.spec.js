import { mount, shallowMount } from "@vue/test-utils";
import NoCreatedItems from "@/components/NoCreatedItems.vue";
import mockRouter from "./mocks/mockRouter.mock";

const mockRoute = {
  params: {
    id: 1,
  },
};

describe("NoCreatedItems.vue", () => {
  it("should render correctly", () => {
    const wrapper = shallowMount(NoCreatedItems);
    expect(wrapper).toMatchSnapshot();
  });

  it("Call To Action Not shown for event invited users", async () => {
    const wrapper = mount(NoCreatedItems, {
      props: {
        invitedUser: true,
      },
    });

    expect(wrapper.find("button").exists()).toBe(false);
  });

  it("Call To Action shown for event creator", async () => {
    const wrapper = mount(NoCreatedItems, {
      props: {
        invitedUser: false,
      },
    });

    expect(wrapper.find("button").isVisible()).toBe(true);
  });

  it("message prop successfully renders to user", async () => {
    const wrapper = mount(NoCreatedItems, {
      props: {
        message: "No Created Items",
      },
    });

    expect(wrapper.find("h4").text()).toBe("No Created Items");
  });

  it("clicking button on component triggers router push", async () => {
    const wrapper = mount(NoCreatedItems, {
      props: {
        message: "No Created Items",
        invitedUser: false,
      },
      global: {
        mocks: {
          $route: mockRoute,
          $router: mockRouter,
        },
      },
    });

    await wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalledTimes(1);
  });
});
