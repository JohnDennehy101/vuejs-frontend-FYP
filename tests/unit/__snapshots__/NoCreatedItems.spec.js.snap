// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoCreatedItems.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        id="no-items-call-to-action-container"
      >
        <i
          class="fas fa-folder-open"
        />
        <h4 />
        <button />
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "callToAction": undefined,
          "callToActionForUser": undefined,
          "display": undefined,
          "guestUser": undefined,
          "hasOwnProperty": [Function],
          "invitedUser": undefined,
          "link": undefined,
          "message": undefined,
          "messageForUser": undefined,
          "noItemsPresent": undefined,
          "routerLink": undefined,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "callToAction": 3,
            "callToActionForUser": 2,
            "constructor": 0,
            "display": 3,
            "guestUser": 2,
            "hasOwnProperty": 4,
            "invitedUser": 3,
            "isVueInstance": 0,
            "link": 2,
            "message": 3,
            "messageForUser": 2,
            "noItemsPresent": 2,
            "routerLink": 3,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "callToAction": undefined,
            "callToActionForUser": undefined,
            "display": undefined,
            "guestUser": undefined,
            "hasOwnProperty": [Function],
            "invitedUser": undefined,
            "link": undefined,
            "message": undefined,
            "messageForUser": undefined,
            "noItemsPresent": undefined,
            "routerLink": undefined,
          },
          "da": null,
          "data": Object {
            "callToActionForUser": undefined,
            "guestUser": undefined,
            "link": undefined,
            "messageForUser": undefined,
            "noItemsPresent": undefined,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "callToAction": undefined,
            "display": undefined,
            "invitedUser": undefined,
            "message": undefined,
            "routerLink": undefined,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "callToAction": Object {},
              "display": Object {},
              "invitedUser": Object {},
              "message": Object {},
              "routerLink": Object {},
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "callToAction": undefined,
            "callToActionForUser": undefined,
            "display": undefined,
            "guestUser": undefined,
            "hasOwnProperty": [Function],
            "invitedUser": undefined,
            "link": undefined,
            "message": undefined,
            "messageForUser": undefined,
            "noItemsPresent": undefined,
            "routerLink": undefined,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <i
                  class="fas fa-folder-open"
                />,
                "key": null,
                "patchFlag": -1,
                "props": Object {
                  "class": "fas fa-folder-open",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "i",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h4 />,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h4",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": null,
                "el": <button />,
                "key": 0,
                "patchFlag": 1,
                "props": Object {
                  "key": 0,
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <h4 />,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "h4",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": null,
                "el": <button />,
                "key": 0,
                "patchFlag": 1,
                "props": Object {
                  "key": 0,
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "dynamicProps": null,
            "el": <div
              id="no-items-call-to-action-container"
            >
              <i
                class="fas fa-folder-open"
              />
              <h4 />
              <button />
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "id": "no-items-call-to-action-container",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "props": Array [
              "display",
              "message",
              "callToAction",
              "routerLink",
              "invitedUser",
            ],
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="no-items-call-to-action-container"
        >
          <i
            class="fas fa-folder-open"
          />
          <h4 />
          <button />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "props": Array [
            "display",
            "message",
            "callToAction",
            "routerLink",
            "invitedUser",
          ],
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="no-items-call-to-action-container"
        >
          <i
            class="fas fa-folder-open"
          />
          <h4 />
          <button />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.26",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "callToAction": undefined,
    "callToActionForUser": undefined,
    "display": undefined,
    "guestUser": undefined,
    "hasOwnProperty": [Function],
    "invitedUser": undefined,
    "link": undefined,
    "message": undefined,
    "messageForUser": undefined,
    "noItemsPresent": undefined,
    "routerLink": undefined,
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <div
    id="no-items-call-to-action-container"
  >
    <i
      class="fas fa-folder-open"
    />
    <h4 />
    <button />
  </div>,
}
`;
